### PR TITLE
Initial code commit

### PR DESCRIPTION
**Pull request by LyleScott**
Let me know what's missing! Give it a whirl.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,66 @@
-# python-fmt
-Python auto formatting with isort and black
+# pyfmt
+
+Python auto formatting using [isort](https://isort.readthedocs.io/en/latest/) and
+[black](https://black.readthedocs.io/en/latest/).
+
+> The arguments to `isort`/`black` are some-what opinionated, but they are what what works for
```
Comment (shawnjoseph) [inline README.md]: "what what"

File: contrib/jenkins/pyfmt.groovy [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/groovy
+
+/*
+ * Install and run pyfmt
+ *
+ * code_dirs is a string representing the paths (space delimted) to pass to pyfmt
+ */
+def call(String code_dirs) {
+    sh "pip3 install --quiet black isort"
+    sh "curl -s -o pyfmt https://raw.githubusercontent.com/GooeeIOT/pyfmt/initial/bin/pyfmt"
```
Comment (rtrajano) [inline contrib/jenkins/pyfmt.groovy]: don't forget to change this to master after the merge

File: contrib/jenkins/pyfmt.groovy [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/groovy
+
+/*
+ * Install and run pyfmt
+ *
+ * code_dirs is a string representing the paths (space delimted) to pass to pyfmt
+ */
+def call(String code_dirs) {
+    sh "pip3 install --quiet black isort"
```
Comment (rtrajano) [inline contrib/jenkins/pyfmt.groovy]: i'm not sure how often these projects get updated but builds/pushes may unexpectedly fail in the future if the checkers implement some new features and the jenkins pipeline command pulls new versions during each build.

Comment (LyleScott) [reply contrib/jenkins/pyfmt.groovy]: good point. i can pin versions

**Review by augustawind — APPROVED** [review]

File: bin/pyfmt [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+if [ -z $@ ]; then
+    to_fmt=${BASE_CODE_DIR}
+else
+    to_fmt=$@
+fi
+
+## Auto arrange imports
+isort \
+    --force-grid-wrap=0 \
+    --line-width=100 \
```
Comment (rtrajano) [inline bin/pyfmt]: should use `$MAX_LINE_LENGTH` here

File: README.md [hunk 5]
```diff
@@ -1,2 +1,66 @@
...
+This is a check only and does not alter your code. Run `pyfmt` to fix.
+
+### Use in a Jenkinsfile
+
+You can add [contrib/jenkins/pyfmt.groovy](contrib/jenkins/pyfmt.groovy) to your Jenkins pipeline
+library to use `pyfmt("paths/to your/code")` in a `Jenkinsfile`.
+
+## Usage
+
+Be advised with projects in a virutal env: `isort` works best when your virtual env is active.
```
Comment (shawnjoseph) [inline README.md]: virtual*

File: bin/pyfmt (hunk 4 above)
Comment (LyleScott) [reply bin/pyfmt]: doh!

**Pull request merged**